pkg/virt-controller/watch/migration.go

```diff
@@ -435,6 +437,30 @@ func (c *MigrationController) createTargetPod(migration *virtv1.VirtualMachineIn
 	return nil
 }
 
+func (c *MigrationController) createPDB(vmi *virtv1.VirtualMachineInstance, vmim *virtv1.VirtualMachineInstanceMigration) error {
+	minAvailable := intstr.FromInt(2)
+	_, err := c.clientset.PolicyV1beta1().PodDisruptionBudgets(vmi.GetNamespace()).Create(context.Background(), &v1beta1.PodDisruptionBudget{
+		ObjectMeta: v1.ObjectMeta{
+			OwnerReferences: []v1.OwnerReference{
+				*v1.NewControllerRef(vmi, virtv1.VirtualMachineInstanceGroupVersionKind),
+			},
+			GenerateName: "kubevirt-disruption-budget-",
```

Comment:
If we use generate name, then we need to use an expecter ref counter similar to what we do with the pod creation. If we don't do this, then  we run the risk of creating multiple PDBs in a loop.

Reply:
I've added a  `migrationPDBSet` which tracks migrations that already have PDB created for, it's basically a very stripped down of the `expectation-like` types, the only reason to not use the existing mechanisms is to not add an informer for PDBs in the migration controller.

The only thing I don't quite like is that items get removed from the set only when the vmim resource is deleted, I'd have preferred for when is completed but I couldn't find an easy way to do it (I probably missed something).

Let me know what you think.

Reply:
the alternative if you don't want to book keep all this is to not use GenerateName, but instead call it something unique and consistent. 

THen when you attempt to create the PBD, ignore the "already exists" error.

Reply:
Sounds a lot easier actually, wii change it. Thanks!

Reply:
PR updated.